Pocket/pocket-ios
# Fix test build failing

## Summary
* This PR fixes some unit tests and improves general testability by adding a `SubscriptionStore` protocol

## Implementation Details
* Add `protocol SubscriptionStore`, a generic subscription store. `PocketSubscriptionStore` is the concrete implementation that handles subscription purchase and retrieval from the App Store.
* Refactor `SearchViewModelTests`

## Test Steps
* Checkout this branch
* Run all unit tests and make sure they pass
* Build and run the app and make sure the purchase flow described in #437 and #421 works as expected
> **Warning**
The same limitations to the purchase flow, as in #421 apply here: testing between launches won't provide reliable results as of now, because we are not syncing the purchase with the backend. If you want to re-test after relaunching, you should delete existing transactions from the `StoreKit` configuration file.

## PR Checklist:
- [x] ~~Added Unit / UI tests~~ - (refactored existing)
